src/weather.cpp

```diff
@@ -166,16 +176,18 @@ weather_sum sum_conditions( const time_point &start, const time_point &end,
         } else {
             tick_size = 1_minutes;
         }
-
+        w_point w = wgen.get_weather( location, t, g->get_seed() );
+        data.averagetemperature += w.temperature;
         weather_type wtype;
         if( g->weather_override == WEATHER_NULL ) {
             wtype = wgen.get_weather_conditions( location, t, g->get_seed() );
         } else {
             wtype = g->weather_override;
         }
         proc_weather_sum( wtype, data, t, tick_size );
+        count += 1;
     }
-
+    data.averagetemperature = static_cast<int>( data.averagetemperature / count );
```

Comment:
This code is also suspicious: `data.averagetemperature / count` follows the integer division rules and can't possibly have a fractional part, casting to `int` is excessive, it's already an `int`. You might want `static_cast<float>(data.averagetemperature) / count` or `averagetemperature` being `float` initially.